bin/header_scraper.py

```diff
@@ -33,15 +34,15 @@ def __init__(self):
 
     def addf(self, content):
```

Comment:
tbh, these `add` functions don't add all that much (punny). See https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters per below. A generic `dict` that you re-fill on each iteration of your HTTP request would suffice too. You're really just using it as a generic container anyway.